src/erlcloud_waf.erl

```diff
@@ -0,0 +1,1188 @@
+        (Config :: aws_config()) ->
+        waf_return_val().
+get_change_token(Config) ->
+    waf_request(Config, "GetChangeToken", []).
+
+
+%%%------------------------------------------------------------------------------
+%%% GetChangeTokenStatus
+%%
+%% http://docs.aws.amazon.com/waf/latest/APIReference/API_GetChangeTokenStatus.html
+%%%------------------------------------------------------------------------------
+-spec get_change_token_status/1 ::
+        (ChangeToken :: string() | binary()) ->
+        waf_return_val().
+get_change_token_status(ChangeToken) ->
+    get_change_token_status(ChangeToken, default_config()).
+
+-spec get_change_token_status/2 ::
+        (ChangeToken:: string() | binary(),
+         Config :: aws_config()) ->
+        waf_return_val().
+get_change_token_status(ChangeToken, Config) ->
+    Json = [{<<"ChangeToken">>, to_binary(ChangeToken)}],
+    waf_request(Config, "GetChangeTokenStatus", Json).
+
+
+%%%------------------------------------------------------------------------------
+%%% GetIPSet
+%%
+%% http://docs.aws.amazon.com/waf/latest/APIReference/API_GetIPSet.html
+%%%------------------------------------------------------------------------------
+-spec get_ip_set/1 ::
+        (IPSetId :: string() | binary()) ->
+        waf_return_val().
+get_ip_set(IPSetId) ->
+    get_ip_set(IPSetId, default_config()).
+
+-spec get_ip_set/2 ::
+        (IPSetId :: string() | binary(),
+         Config :: aws_config()) ->
+        waf_return_val().
+get_ip_set(IPSetId, Config) ->
+    Json = [{<<"IPSetId">>, to_binary(IPSetId)}],
+    waf_request(Config, "GetIPSet", Json).
+
+
+%%%------------------------------------------------------------------------------
+%%% GetRule
+%%
+%% http://docs.aws.amazon.com/waf/latest/APIReference/API_GetRule.html
+%%%------------------------------------------------------------------------------
+-spec get_rule/1 ::
+        (RuleId :: string() | binary()) ->
+        waf_return_val().
+get_rule(RuleId) ->
+    get_rule(RuleId, default_config()).
+
+-spec get_rule/2 ::
+        (RuleId :: string() | binary(),
+         Config :: aws_config()) ->
+        waf_return_val().
+get_rule(RuleId, Config) ->
+    Json = [{<<"RuleId">>, to_binary(RuleId)}],
+    waf_request(Config, "GetRule", Json).
+
+
+%%%------------------------------------------------------------------------------
+%%% GetSampledRequests
+%%
+%% http://docs.aws.amazon.com/waf/latest/APIReference/API_GetSampledRequests.html
+%%%------------------------------------------------------------------------------
+-spec get_sampled_requests/5 ::
+        (MaxItems :: pos_integer(),
+         RuleId :: string() | binary(),
+         StartTime :: pos_integer(),
+         EndTime :: pos_integer(),
+         WebAclId :: string() | binary()) ->
+        waf_return_val().
+get_sampled_requests(MaxItems, RuleId, StartTime, EndTime, WebAclId) ->
+    get_sampled_requests(MaxItems, RuleId, StartTime, EndTime, WebAclId, default_config()).
```

Comment:
`get_sampled_requests(WebAclId, RuleId, StartTime, EndTime, MaxItems)`

http://docs.aws.amazon.com/waf/latest/APIReference/API_GetSampledRequests.html#AWS%20WAF-GetSampledRequests-request-MaxItems
MaxItems
The number of requests that you want AWS WAF to return from among the first 5,000 requests that your AWS resource received **during the time range**.

Therefore, `MaxItems` is after `TimeWindow`.